### PR TITLE
Added Basic Search

### PR DESCRIPTION
**Pull request by modmuss50**
Adds a Search window that can be opened via the menu item or Shift + Space

Currently very basic, only allows searching of outer class names. I made use of a libary to help with the searching as just doing a string.contains was not working well. This could be improved upon, if anyone has any suggestions let me know.

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -99,6 +99,7 @@ dependencies {
 
     application name: "darcula", version: "1.0.0"
     application 'de.sciss:syntaxpane:1.1.+'
+	application 'me.xdrop:fuzzywuzzy:1.2.0'
```
Comment (Juuxel) [inline build.gradle]: The indentation is pretty fuzzy-wuzzy. 😅

File: src/main/java/cuchaz/enigma/gui/dialog/SearchDialog.java [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+		searchField.requestFocusInWindow();
+	}
+
+	private void openSelected(){
+		close();
+		if(classList.isSelectionEmpty()){
+			return;
+		}
+		deobfClasses.stream()
+			.filter(classEntry -> !classEntry.isInnerClass())
```
Comment (gegy1000) [inline src/main/java/cuchaz/enigma/gui/dialog/SearchDialog.java]: This check needed? Inner classes are already removed from the list

File: build.gradle (hunk 1 above)
Comment (modmuss50) [reply build.gradle]: haha, yeah. will fix.

File: src/main/java/cuchaz/enigma/gui/dialog/SearchDialog.java (hunk 2 above)
Comment (modmuss50) [reply src/main/java/cuchaz/enigma/gui/dialog/SearchDialog.java]: Nope, will fix

File: src/main/java/cuchaz/enigma/gui/dialog/SearchDialog.java [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+		pane.add(panel, BorderLayout.CENTER);
+	}
+
+	//Updates the list of class names
+	private void updateList() {
+		DefaultListModel<String> listModel = new DefaultListModel<>();
+
+		//Basic search using the Fuzzy libary
+		//TODO improve on this, to not just work from string and to keep the ClassEntry
+		List<ExtractedResult> results = FuzzySearch.extractTop(searchField.getText(), deobfClasses.stream().map(ClassEntry::getSimpleName).collect(Collectors.toList()), 25);
```
[the remark this answers is not in the document]
Comment (modmuss50) [reply src/main/java/cuchaz/enigma/gui/dialog/SearchDialog.java]: I think is is an improvement that can be made later.

The current lib for searching only works on strings as far as I can tell, the improvement that I am suggesting for the future would be to come up with a search algorithm that works on ClassEntry's that could be used to display more data in the search results such as package names.

I think for now this is best left alone. If you want I can remove the TODO and just keep the comment.

**Comment by gegy1000** [comment]
Is this ready to merge, then?

**Comment by modmuss50** [comment]
Yes should be

**Comment by asiekierka** [comment]
Ping?

**Pull request merged**